File: src/main/java/com/example/stopgap/MainConfig.java
package com.example.stopgap;

import com.example.stopgap.generator.GeneratorConfig;
import com.example.stopgap.instanceregistry.InstanceRegistry;

final class MainConfig {
    private MainConfig() {
    }

    static void setup(final InstanceRegistry registry) {
        registry.registerForType(MainEndpoint.class, MainConfig::mainEndpoint);
        GeneratorConfig.setup(registry);
    }

    private static MainEndpoint mainEndpoint(final InstanceRegistry registry) {
        return new MainEndpoint();
    }
}

File: app/src/main/kotlin/com/example/stopgap/MainConfig.kt
package com.example.stopgap

import dev.sku20.ir.Creates
import dev.sku20.ir.InstanceRegistry
import io.helidon.config.Config

object MainConfig {

    @Creates
    fun config(registry: InstanceRegistry): Config {
        val config = Config.create()
        return config
    }

    fun mainEndpoint(registry: InstanceRegistry): MainEndpoint {
        return MainEndpoint()
    }

}
